backend/src/main/kotlin/blue/mild/covid/vaxx/service/PatientService.kt

```diff
@@ -1,129 +1,153 @@
+import blue.mild.covid.vaxx.dto.request.PatientUpdateDtoIn
 import blue.mild.covid.vaxx.dto.response.PatientDtoOut
 import blue.mild.covid.vaxx.dto.response.PatientRegisteredDtoOut
 import blue.mild.covid.vaxx.error.entityNotFound
+import mu.KLogging
+import org.jetbrains.exposed.sql.Column
 import org.jetbrains.exposed.sql.Op
-import org.jetbrains.exposed.sql.ResultRow
-import org.jetbrains.exposed.sql.SqlExpressionBuilder
-import org.jetbrains.exposed.sql.batchInsert
-import org.jetbrains.exposed.sql.deleteWhere
-import org.jetbrains.exposed.sql.insert
-import org.jetbrains.exposed.sql.leftJoin
-import org.jetbrains.exposed.sql.select
-import org.jetbrains.exposed.sql.selectAll
+import org.jetbrains.exposed.sql.and
 import org.jetbrains.exposed.sql.transactions.experimental.newSuspendedTransaction
-import pw.forst.tools.katlib.TimeProvider
-import pw.forst.tools.katlib.toUuid
-import java.time.Instant
+import pw.forst.tools.katlib.applyIf
+import pw.forst.tools.katlib.whenFalse
 import java.util.UUID
 
 class PatientService(
+    private val patientRepository: PatientRepository,
     private val validationService: ValidationService,
-    private val instantTimeProvider: TimeProvider<Instant>,
     private val entityIdProvider: EntityIdProvider
 ) {
 
-    suspend fun getPatientById(patientId: UUID): PatientDtoOut = newSuspendedTransaction {
-        val data = Patient
-            .leftJoin(Answer, { id }, { Answer.patientId })
-            .select { Patient.id eq patientId.toString() }
-            .toList() // eager fetch all data from the database
-
-        val answers = data.map(::mapAnswer)
-
-        data.firstOrNull()
-            ?.let { mapPatient(it, answers) }
-            ?: throw entityNotFound<Patient>(Patient::id, patientId)
-    }
+    private companion object : KLogging()
+
+    /**
+     * Returns patient with given ID.
+     */
+    suspend fun getPatientById(patientId: UUID): PatientDtoOut =
+        patientRepository.getAndMapPatientsBy { Patient.id eq patientId.toString() }
+            .singleOrNull() ?: throw entityNotFound<Patient>(Patient::id, patientId)
+
+    /**
+     * Returns single patient with given personal number or throws exception.
+     */
+    suspend fun getPatientsByPersonalNumber(patientPersonalNumber: String): PatientDtoOut =
+        patientRepository.getAndMapPatientsBy {
+            Patient.personalNumber eq normalizePersonalNumber(patientPersonalNumber)
+        }.singleOrNull() ?: throw entityNotFound<Patient>(Patient::personalNumber, patientPersonalNumber)
+
+    /**
+     * Filters the database with the conjunction (and clause) of the given properties.
+     */
+    suspend fun getPatientsByConjunctionOf(
+        email: String? = null,
+        phoneNumber: String? = null,
+        vaccinated: Boolean? = null
+    ): List<PatientDtoOut> =
+        patientRepository.getAndMapPatientsBy {
+            Op.TRUE
+                .andWithIfNotEmpty(email?.trim(), Patient.email)
```

Comment:
Tady bacha na male/velka pismena. Ale myslim, ze jsem do DB daval ten datovy typ, kteremu to je jedno. Ted, jestli je pouzitej. ITEXT myslim.